core/storage/user/gae_models.py

```diff
@@ -3011,3 +3011,139 @@ def get_export_policy(cls) -> Dict[str, base_models.EXPORT_POLICY]:
         """
         empty_dict: Dict[str, base_models.EXPORT_POLICY] = {}
         return dict(super(cls, cls).get_export_policy(), **empty_dict)
+
+
+class LearnerGroupUserModel(base_models.BaseModel):
+    """Model for storing user's learner groups related data.
+
+    Instances of this class are keyed by the user id.
+    """
+
+    # List of learner group ids which the user has been invited to join.
+    invited_to_learner_groups_ids = (
+        datastore_services.StringProperty(repeated=True, indexed=True))
+    # List of learner group ids which the user has joined as a student.
+    student_of_learner_groups_ids = (
+        datastore_services.StringProperty(repeated=True, indexed=True))
+    # List of ProgressSharingPermission dicts, which is defined in
+    # user_domain.py, each dict corresponds to a learner group and
+    # has stores progress sharing permission status for that group.
+    progress_sharing_permissions = (
+        datastore_services.JsonProperty(repeated=True, indexed=False))
+
+    @staticmethod
+    def get_deletion_policy() -> base_models.DELETION_POLICY:
+        """Model contains data to delete corresponding to a user: id field."""
+        return base_models.DELETION_POLICY.DELETE
+
+    @classmethod
+    def has_reference_to_user_id(cls, user_id: str) -> bool:
+        """Check whether LearnerGroupUserModel exists for the given user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether any models refer to the given user ID.
+        """
+        return cls.get_by_id(user_id) is not None
+
+    @classmethod
+    def apply_deletion_policy(cls, user_id: str) -> None:
+        """Delete instances of LearnerGroupUserModel for the user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be deleted.
+        """
+        cls.delete_by_id(user_id)
+
+    @classmethod
+    def export_data(
+        cls,
+        user_id: str
+    ) -> Dict[str, Union[List[str], List[Dict[str, Union[bool, str]]], None]]:
```

Comment:
Since we know the keys of the response dict, I think its better to define its type using `TypedDict`?

Reply:
Done!